### PR TITLE
[O2B-1043] Add pre-filled options for time range filter

### PR DESCRIPTION
**Pull request by martinboulais**
#### I have a JIRA ticket
- [X] branch and/or PR name(s) include(s) JIRA ID
- [X] issue has "Fix version" assigned
- [X] issue "Status" is set to "In review"
- [X] PR labels are selected

Notable changes for users:
- Added pre-fill options to the time filter

### CODE REVIEW

**Review by xsalonx — CHANGES_REQUESTED** [review]
I think there it can be a bit improved:

1. After selecting e.g. year 2022, still, there is title `Efficiency - 2023`, what is a bit misleading (the same for the page `lhc-fill-overview`)
2. After setting a week and changing to month that the given week dosen't belong to, the week number remains the same, for me it should not.
3. Maybe it's intended because of transforming to locale time, to be confirmed, if I choose e.g. some year I can see it is from 01:00 am to 01:00 am next year, should it be like this?
4. I suppose that months grid, present last twelve months. If so current solution with putting year at the right side of month name might be not very nice after reaching half of the next year. I would propose to mark year they belong to by setting some colour of background for particular month (or text, to be considered). Then years' tabs above might be styled as well with appropriate colours. 
Screenshot 2023-11-22 at 16 58 32
Screenshot 2023-11-22 at 16 56 22
Screenshot 2023-11-22 at 16 56 02

**Review by xsalonx — CHANGES_REQUESTED** [review]

**Review by xsalonx — CHANGES_REQUESTED** [review]
Also, it's not very elegant that the date selection grid is 'leaky', hoovering affects underlying chart
image

**Review by xsalonx — DISMISSED** [review]

File: lib/public/components/Filters/common/filters/timeRangeFilter.js [hunk 1]
```diff
@@ -13,7 +13,72 @@
...
+    const currentMonthIndex = now.getUTCMonth();
+    const currentYear = now.getFullYear();
+
+    return h('.month-options', new Array(MONTH_NAMES.length).fill(0).map((_, index) => {
+        const monthIndex = (index + currentMonthIndex + 1) % MONTH_NAMES.length;
+        const year = monthIndex > currentMonthIndex ? currentYear - 1 : currentYear;
+        let label = MONTH_NAMES[monthIndex];
+        if (currentYear !== year) {
+            label += ` (${year})`;
+        }
```
[the remark this answers is not in the document]
Comment (martinboulais) [reply lib/public/components/Filters/common/filters/timeRangeFilter.js:76]: I understand the point of view, I propose to merge as is for now and we discuss this further in a future PR

File: lib/public/components/Filters/common/filters/timeRangeFilter.js [hunk 2]
```diff
@@ -13,7 +13,72 @@
...
+            },
+            year,
+        );
+    }));
+};
+
+/**
+ * Return a pre-defined list of months period selectors
+ *
+ * @param {function} onChange function called with the clicked period
```
[the remark this answers is not in the document]
Comment (martinboulais) [reply lib/public/components/Filters/common/filters/timeRangeFilter.js]: Agree, done

File: lib/public/components/Filters/common/filters/timeRangeFilter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/components/Filters/common/filters/timeRangeFilter.js:76]: agree

**Pull request merged**